intermittent `error="failed to call PTRACE_CONT (pid=4783) no such process"`

https://github.com/AkihiroSuda/gomodjail/actions/runs/14613096439/job/40995044657?pr=42

```
+ gomodjail pack --go-mod=go.mod ./victim
time=2025-04-23T08:06:53.474Z level=INFO msg="Creating a self-extract archive" file=victim.gomodjail
+ ./victim.gomodjail
time=2025-04-23T08:06:53.491Z level=WARN msg=***Blocked*** pid=4777 exe=/tmp/gomodjail-596246853/victim syscall=fcntl entry=/home/runner/work/gomodjail/gomodjail/examples/poisoned/poisoned.go:24:github.com/AkihiroSuda/gomodjail/examples/poisoned.Add module=github.com/AkihiroSuda/gomodjail/examples/poisoned
time=2025-04-23T08:06:53.491Z level=WARN msg=***Blocked*** pid=4777 exe=/tmp/gomodjail-596246853/victim syscall=fcntl entry=/home/runner/work/gomodjail/gomodjail/examples/poisoned/poisoned.go:24:github.com/AkihiroSuda/gomodjail/examples/poisoned.Add module=github.com/AkihiroSuda/gomodjail/examples/poisoned
time=2025-04-23T08:06:53.491Z level=WARN msg=***Blocked*** pid=4777 exe=/tmp/gomodjail-596246853/victim syscall=fcntl entry=/home/runner/work/gomodjail/gomodjail/examples/poisoned/poisoned.go:24:github.com/AkihiroSuda/gomodjail/examples/poisoned.Add module=github.com/AkihiroSuda/gomodjail/examples/poisoned
time=2025-04-23T08:06:53.491Z level=WARN msg=***Blocked*** pid=4777 exe=/tmp/gomodjail-596246853/victim syscall=fcntl entry=/home/runner/work/gomodjail/gomodjail/examples/poisoned/poisoned.go:24:github.com/AkihiroSuda/gomodjail/examples/poisoned.Add module=github.com/AkihiroSuda/gomodjail/examples/poisoned
time=2025-04-23T08:06:53.492Z level=WARN msg=***Blocked*** pid=4777 exe=/tmp/gomodjail-596246853/victim syscall=pidfd_open entry=/home/runner/work/gomodjail/gomodjail/examples/poisoned/poisoned.go:24:github.com/AkihiroSuda/gomodjail/examples/poisoned.Add module=github.com/AkihiroSuda/gomodjail/examples/poisoned
time=2025-04-23T08:06:53.492Z level=WARN msg=***Blocked*** pid=4777 exe=/tmp/gomodjail-596246853/victim syscall=waitid entry=/home/runner/work/gomodjail/gomodjail/examples/poisoned/poisoned.go:24:github.com/AkihiroSuda/gomodjail/examples/poisoned.Add module=github.com/AkihiroSuda/gomodjail/examples/poisoned
time=2025-04-23T08:06:53.492Z level=WARN msg=***Blocked*** pid=4777 exe=/tmp/gomodjail-596246853/victim syscall=fcntl entry=/home/runner/work/gomodjail/gomodjail/examples/poisoned/poisoned.go:24:github.com/AkihiroSuda/gomodjail/examples/poisoned.Add module=github.com/AkihiroSuda/gomodjail/examples/poisoned
time=2025-04-23T08:06:53.492Z level=WARN msg=***Blocked*** pid=4788 exe=/tmp/gomodjail-596246853/victim syscall=fcntl entry=/home/runner/work/gomodjail/gomodjail/examples/poisoned/poisoned.go:24:github.com/AkihiroSuda/gomodjail/examples/poisoned.Add module=github.com/AkihiroSuda/gomodjail/examples/poisoned
time=2025-04-23T08:06:53.492Z level=WARN msg=***Blocked*** pid=4788 exe=/tmp/gomodjail-596246853/victim syscall=fcntl entry=/home/runner/work/gomodjail/gomodjail/examples/poisoned/poisoned.go:24:github.com/AkihiroSuda/gomodjail/examples/poisoned.Add module=github.com/AkihiroSuda/gomodjail/examples/poisoned
time=2025-04-23T08:06:53.492Z level=WARN msg=***Blocked*** pid=4788 exe=/tmp/gomodjail-596246853/victim syscall=execve entry=/home/runner/work/gomodjail/gomodjail/examples/poisoned/poisoned.go:24:github.com/AkihiroSuda/gomodjail/examples/poisoned.Add module=github.com/AkihiroSuda/gomodjail/examples/poisoned
time=2025-04-23T08:06:53.492Z level=ERROR msg="exiting with an error" error="failed to call PTRACE_CONT (pid=4783) no such process"
42 + 43 = 85
Error: Process completed with exit code 1.
```

--- a/pkg/profile/profile.go
+++ b/pkg/profile/profile.go
@@ -1,7 +1,6 @@
 package profile
 
 import (
-	"errors"
 	"fmt"
 	"log/slog"
 	"slices"
@@ -38,7 +37,7 @@ type Profile struct {
 
 func (p *Profile) Validate() error {
 	if p.Module == "" {
-		return errors.New("no module was specified")
+		slog.Warn("No module was specified")
 	}
 	if len(p.Modules) == 0 {
 		slog.Warn("No policy was specified")

--- a/pkg/tracer/tracer_linux.go
+++ b/pkg/tracer/tracer_linux.go
@@ -96,46 +96,54 @@ func (tracer *tracer) Trace() error {
 	if err := unix.PtraceCont(wPid, 0); err != nil {
 		return fmt.Errorf("failed to call PTRACE_CONT (pid=%d) %w", wPid, err)
 	}
+
 	for {
-		var ws unix.WaitStatus
-		wPid, err = unix.Wait4(-1*pGid, &ws, unix.WALL, nil)
-		if err != nil {
+		// trace may call os.Exit
+		if err = tracer.trace(pGid); err != nil {
 			return err
 		}
-		switch {
-		case ws.Exited():
-			exitStatus := ws.ExitStatus()
-			if wPid == tracer.cmd.Process.Pid {
-				if exitStatus == 0 {
-					slog.Debug("exiting")
-				} else {
-					slog.Error("exiting with non-zero status", "status", exitStatus)
-				}
-				os.Exit(exitStatus)
-				return nil
-			}
-			continue
-		}
-		switch uint32(ws) >> 8 {
-		case uint32(unix.SIGTRAP) | (unix.PTRACE_EVENT_SECCOMP << 8):
-			var regs regs.Regs
-			if err = unix.PtraceGetRegs(wPid, &regs.PtraceRegs); err != nil {
-				return fmt.Errorf("failed to read registers for %d: %w", wPid, err)
-			}
-			if err = tracer.handleSyscall(wPid, &regs); err != nil {
-				slog.Debug("failed to handle syscall", "pid", wPid, "syscall", regs.Syscall(), "error", err)
+	}
+}
+
+func (tracer *tracer) trace(pGid int) error {
+	var ws unix.WaitStatus
+	wPid, err := unix.Wait4(-1*pGid, &ws, unix.WALL, nil)
+	if err != nil {
+		return err
+	}
+	switch {
+	case ws.Exited():
+		exitStatus := ws.ExitStatus()
+		if wPid == tracer.cmd.Process.Pid {
+			if exitStatus == 0 {
+				slog.Debug("exiting")
 			} else {
-				if regs.Modified {
-					if err = unix.PtraceSetRegs(wPid, &regs.PtraceRegs); err != nil {
-						return fmt.Errorf("failed to set registers for %d: %w", wPid, err)
-					}
+				slog.Error("exiting with non-zero status", "status", exitStatus)
+			}
+			os.Exit(exitStatus)
+		}
+		return nil
+	}
+	switch uint32(ws) >> 8 {
+	case uint32(unix.SIGTRAP) | (unix.PTRACE_EVENT_SECCOMP << 8):
+		var regs regs.Regs
+		if err = unix.PtraceGetRegs(wPid, &regs.PtraceRegs); err != nil {
+			return fmt.Errorf("failed to read registers for %d: %w", wPid, err)
+		}
+		if err = tracer.handleSyscall(wPid, &regs); err != nil {
+			slog.Debug("failed to handle syscall", "pid", wPid, "syscall", regs.Syscall(), "error", err)
+		} else {
+			if regs.Modified {
+				if err = unix.PtraceSetRegs(wPid, &regs.PtraceRegs); err != nil {
+					return fmt.Errorf("failed to set registers for %d: %w", wPid, err)
 				}
 			}
-		}
-		if err := unix.PtraceCont(wPid, 0); err != nil {
-			return fmt.Errorf("failed to call PTRACE_CONT (pid=%d) %w", wPid, err)
 		}
 	}
+	if err := unix.PtraceCont(wPid, 0); err != nil {
+		return fmt.Errorf("failed to call PTRACE_CONT (pid=%d) %w", wPid, err)
+	}
+	return nil
 }
 
 func (tracer *tracer) handleSyscall(pid int, regs *regs.Regs) error {

--- a/pkg/tracer/tracer_linux.go
+++ b/pkg/tracer/tracer_linux.go
@@ -100,7 +100,11 @@ func (tracer *tracer) Trace() error {
 	for {
 		// trace may call os.Exit
 		if err = tracer.trace(pGid); err != nil {
-			return err
+			if errors.Is(err, unix.ESRCH) {
+				slog.Debug("ESRCH", "error", err)
+			} else {
+				return err
+			}
 		}
 	}
 }

